pkg/kv/kvclient/kvcoord/dist_sender_rangefeed.go

```diff
@@ -87,7 +88,7 @@ func (ds *DistSender) RangeFeed(
 	spans []roachpb.Span,
 	startAfter hlc.Timestamp, // exclusive
 	withDiff bool,
-	eventCh chan<- *roachpb.RangeFeedEvent,
+	eventCh chan<- *kvclient.RangeFeedEnvelope,
```

Comment:
Wouldn't we then have to migrate all callers over to this new `ConsumeRangeFeed` method anyway, which is presumably an even larger change since it'd introduce callbacks everywhere?

I don't have a strong opinion on channels vs callbacks -- slight preference for channels because they can be combined via `select`. If you generally prefer callbacks on the client side then that's fine with me, since CDC is the primary consumer anyway.

I'm not sure if it's the best use of @gh-casper's time for him to keep rewriting this though. If he'd rather keep it as-is and work on something more important then I'd be +1 on that.